src/index.ts

```diff
@@ -181,66 +184,66 @@ function startApp(appSchema, path: string) {
     server.applyMiddleware({ app, path })
   } else {
     const graphqlHTTP = require("express-graphql")
-    app.use(
-      graphqlHTTP((req, res, params) => {
-        const accessToken = req.headers["x-access-token"] as string | undefined
-        const appToken = req.headers["x-xapp-token"] as string | undefined
-        const userID = req.headers["x-user-id"] as string | undefined
-        const timezone = req.headers["x-timezone"] as string | undefined
-        const userAgent = req.headers["user-agent"]
-
-        const { requestIDs } = res.locals
-        const requestID = requestIDs.requestID
-
-        if (enableRequestLogging) {
-          fetchLoggerSetup(requestID)
-        }
-
-        // Accepts a tz database timezone string. See http://www.iana.org/time-zones,
-        // https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
-        let defaultTimezone
-        if (timezone && moment.tz.zone(timezone)) {
-          defaultTimezone = timezone
-        }
-
-        const loaders = createLoaders(accessToken, userID, {
-          requestIDs,
-          userAgent,
-          appToken,
-        })
-
-        const context: ResolverContext = {
-          accessToken,
-          userID,
-          defaultTimezone,
-          ...loaders,
-          // For stitching purposes
-          exchangeSchema,
-          requestIDs,
-          userAgent,
-        }
+    const graphqlServer = graphqlHTTP((req, res, params) => {
+      const accessToken = req.headers["x-access-token"] as string | undefined
+      const appToken = req.headers["x-xapp-token"] as string | undefined
+      const userID = req.headers["x-user-id"] as string | undefined
+      const timezone = req.headers["x-timezone"] as string | undefined
+      const userAgent = req.headers["user-agent"]
+
+      const { requestIDs = {} } = res.locals ?? {}
```

Comment:
This line contains the only change here. `res.locals` is undefined when batched.

Reply:
The rest of this diff is just moving the middleware instantiation into a variable `graphqlServer` as it's mounted directly, and then inside the batch middleware.

Reply:
I don't know if this has any potential impact on how we do logging.

Reply:
Are there any other impacts to `res.locals` being undefined?

Reply:
Sorry, I think rather than `res.locals` being undefined. It's just `res.locals.requestIDs`. There's a middleware that shovels them onto the request for logging. And with batch requests, there are no request IDs.

> As of v4.0.0, the batch middleware utilizing request IDs in queries and corresponding results has been renamed legacyBatchMiddleware. The legacy middleware included a request ID with each query included in the batch and expected the server to return each result with the corresponding request ID. The new batchMiddleware simply expects results be returned in the same order as the batched queries.

Reply:
(But yeah it'd be more appropriate to make this change within that middleware)

Reply:
OK so it actually was being undefined and it was due to a small bug in the batch middleware. Patched the issue along with exporting the type in https://github.com/relay-tools/react-relay-network-modern/pull/116